### PR TITLE
E_CLASSROOM-276 [FE/BE] Accounts List Page

### PR DESCRIPTION
**Pull request by Teraaay**
### Issue Link

https://framgiaph.backlog.com/view/E_CLASSROOM-276

### Definition of Done
- [x] Dynamic display of all admin accounts
- [x] Pagination is working
- [x] Seeder command is working
- [x] 'Add an Admin' button should display but no functionality yet

### Related PR

BE Link: https://github.com/framgia/sph-classroom-els-be/pull/66

#### Main note
- Run the seeders command mentioned in the BE PR to add more admin users to test pagination functionality, before checking FE PR.
#### Pages Affected

- ADMIN ACCOUNTS PAGE: http://localhost:3003/admin/users

### CODE REVIEW

File: src/api/Admin/index.js [hunk 1]
```diff
@@ -47,6 +47,17 @@ const AdminApi = {
       }
     };
 
+    return API.request(options);
+  },
+
+  getAdminAccounts: ({ ...params }) => {
+    const options = {
+      method: 'GET',
+      url: '/admin/admins',
```
Comment (karlolee50) [inline src/api/Admin/index.js]: change this URL once you also update the backend URL

File: src/components/NavigationSideBar/index.js [hunk 2]
```diff
@@ -86,7 +86,7 @@ const NavigationSideBar = () => {
           </Nav.Link>
         </LinkContainer>
         <LinkContainer
-          to="/admin"
+          to="/admin/admins"
```
Comment (karlolee50) [inline src/components/NavigationSideBar/index.js]: this one also

File: src/pages/admin/Admin/AdminList/index.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    AdminApi.getAdminAccounts({ page })
+      .then(({ data }) => {
+        setAdminAccounts(data.data);
+        setPerPage(data.per_page);
+        setTotalItems(data.total);
+        setLastPage(data.last_page);
+      })
+      .catch(() =>
+        toast('Error', 'There was an error getting the list of admin accounts.')
+      );
```
Comment (karlolee50) [inline src/pages/admin/Admin/AdminList/index.js:54]: transfer into a 'load' function

File: src/routes/index.js [hunk 4]
```diff
@@ -113,6 +114,8 @@ const Routes = () => {
         component={AdminEditPassword}
       ></AdminRoute>
 
+      <AdminRoute path="/admin/admins" exact component={AdminList}></AdminRoute>
```
Comment (karlolee50) [inline src/routes/index.js]: update URL here also

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 10 minutes
Comment/s: 2

**Comment by ErickDoroy** [comment]
## Working
- [x] Dynamic display of all admin accounts
- [x] Pagination is working
- [x] 'Add an Admin' button should display but no functionality yet
## Comment
- [x] Please delete edit area it must be the same as figma
accountdelete

--------------
- Total Test Time: 0.75 hr
- Total Iterations: 1
- Total Comments: 1

**Comment by ErickDoroy** [comment]
## APPROVE #130 
-----------
- Total Test Time: 0.75 hr
- Total Iterations: 1
- Total Comments: 1

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**